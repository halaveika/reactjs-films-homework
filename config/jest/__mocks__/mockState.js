const mockState = {
  content: {
    results: [{
      id: 2412412,
      title: 'Fight club',
      genre_ids: [13, 20, 21],
      vote_average: 8,
      poster_path: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    },
    {
      id: 41412,
      title: 'Sopranos',
      genre_ids: [13, 18],
      vote_average: 10,
      poster_path: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry',
    }],
    page: 1,
    total_pages: 12,
    total_results: 55,
    genres_array: [{
      id: 13,
      name: 'Adventure',
    }, {
      id: 18,
      name: 'Drama',
    }, {
      id: 20,
      name: 'Family',
    }, {
      id: 21,
      name: 'Fantasy',
    }],
    video_url: 'test_url',
    details: {
      id: 2412412,
      title: 'Fight club',
      runtime: 120,
      vote_average: 10,
      poster: 'image_path',
      overview: 'Lorem Ipsum is simply dummy text of the printing',
    },
    isInitialisated: false,
  },
};

export default mockState
