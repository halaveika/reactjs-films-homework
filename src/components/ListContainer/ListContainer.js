import { connect } from 'react-redux';
import MovieList from '../../pages/MovieList';
import { getSearchList, getVideo, getMoviesGenre,getLoadingSelector, getInitialisation, getFilter, getPageSelector, getTotalPageSelector } from '../../modules/content/contentSelectors';
import { GetVideoUrl, getDetails, getTrending, getTopRated, getUpcoming,getInitialisated,GetGenres, setFilter, setCurrentPage} from '../../modules/content/contentActions';

export const mapStateToProps = (state) => ({
  items: getSearchList(state),
  video: getVideo(state),
  genres: getMoviesGenre(state),
  isLoading: getLoadingSelector(state),
  isInitialisated: getInitialisation(state),
  filter: getFilter(state),
  page: getPageSelector(state),
  total_page: getTotalPageSelector(state)
});

const ListContainer = connect(mapStateToProps,
  { handleVideo: GetVideoUrl, getDetails, getTrending, getTopRated, getUpcoming,getInitialisated,GetGenres, setFilter, setCurrentPage})(MovieList);

export default ListContainer;
