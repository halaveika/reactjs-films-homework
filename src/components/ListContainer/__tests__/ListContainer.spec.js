import { mapStateToProps } from '../ListContainer';
import mockState from '../../../../config/jest/__mocks__/mockState';

describe('ListContainer', () => {
  it('should show mapStateToProps value', () => {
    expect(mapStateToProps(mockState)).toMatchSnapshot();
  });
});
