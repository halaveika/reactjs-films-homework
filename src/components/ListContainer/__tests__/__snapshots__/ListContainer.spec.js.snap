// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer should show mapStateToProps value 1`] = `
Object {
  "filter": "TRENDING",
  "genre": "",
  "genres": Array [
    Object {
      "id": 13,
      "name": "Adventure",
    },
    Object {
      "id": 18,
      "name": "Drama",
    },
    Object {
      "id": 20,
      "name": "Family",
    },
    Object {
      "id": 21,
      "name": "Fantasy",
    },
  ],
  "isLoading": true,
  "isRow": true,
  "items": Array [
    Object {
      "genres": Array [
        "Adventure",
        "Family",
        "Fantasy",
      ],
      "id": 2412412,
      "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
      "poster": "https://image.tmdb.org/t/p/w500/image_path",
      "title": "Fight club",
      "vote_average": 8,
    },
    Object {
      "genres": Array [
        "Adventure",
        "Drama",
      ],
      "id": 41412,
      "overview": "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
      "poster": "https://image.tmdb.org/t/p/w500/image_path",
      "title": "Sopranos",
      "vote_average": 10,
    },
  ],
  "page": 1,
  "pageSize": 20,
  "searchValue": "",
  "total_results": 55,
}
`;
